pkg/operators/wasm/wasm.go

```diff
@@ -101,6 +101,8 @@ func (w *wasmOperator) InstantiateImageOperator(
 }
 
 type wasmOperatorInstance struct {
+	ctx       context.Context
+	cancel    func()
```

Comment:
Could it be named preStartCtx and preStartCancel?

I am not sure I understand why it is needed, despite the explanation below. What does `callGuestFunction` do with the context?

Reply:
The issue is the following: gadgetCtx.Context() was passed to PreStart() where many gadgets subscribed to a data source, then this context was [passed](https://github.com/inspektor-gadget/inspektor-gadget/blob/8e1aecb422d023757650ec0bb61ecc7d91b8c29b/pkg/operators/wasm/datasource.go#L295-L317) to the callback. Then, when gadgetCtx.Context()  was cancelled, the subscriptions didn't get invoked because ctx in `_, err := i.dataSourceCallback.Call(ctx`  was cancelled.